kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/util/PppAuthSecrets.java

```diff
@@ -231,63 +229,32 @@ private void writeToFile() throws Exception {
      *             for indexing problems
      */
     public void removeEntry(String type, String value) throws Exception {
-        if (type.compareTo("provider") == 0) {
-            for (int i = 0; i < this.providers.size(); ++i) {
-                if (this.providers.get(i).compareTo(value) == 0) {
-                    try {
-                        this.removeEntry(i);
-                    } catch (Exception e) {
-                        throw e;
-                    }
-                }
-            }
-        } else if (type.compareTo("client") == 0) {
-            for (int i = 0; i < this.clients.size(); ++i) {
-                if (this.clients.get(i).compareTo(value) == 0) {
-                    try {
-                        this.removeEntry(i);
-                    } catch (Exception e) {
-                        throw e;
-                    }
-                }
-            }
-        } else if (type.compareTo("server") == 0) {
-            for (int i = 0; i < this.servers.size(); ++i) {
-                if (this.servers.get(i).compareTo(value) == 0) {
-                    try {
-                        this.removeEntry(i);
-                    } catch (Exception e) {
-                        throw e;
-                    }
-                }
-            }
-        } else if (type.compareTo("secret") == 0) {
-            for (int i = 0; i < this.secrets.size(); ++i) {
-                if (this.secrets.get(i).compareTo(value) == 0) {
-                    try {
-                        this.removeEntry(i);
-                    } catch (Exception e) {
-                        throw e;
-                    }
-                }
-            }
-        } else if (type.compareTo("ipAddress") == 0) {
-            for (int i = 0; i < this.ipAddresses.size(); ++i) {
-                if (this.ipAddresses.get(i).compareTo(value) == 0) {
-                    try {
-                        this.removeEntry(i);
-                    } catch (Exception e) {
-                        throw e;
-                    }
-                }
-            }
+        if ("provider".equals(type)) {
```

Comment:
I've not done a search on this, but do you think it would be worth moving those strings to an enum? Maybe we can reduce duplication and error possibilities.